proto/dwtools/abase/l0.test/Long.test.s

```diff
@@ -13427,14 +13427,80 @@ function longFromRange( test )
   });
 
 };
+
 //
 
-function longFromRange1( test )
+function longFromRangeWithStep( test )
 {
+  test.case = 'equal numders of range';
+  var got = _.longFromRangeWithStep( [ 1, 1 ], 2 );
+  var expected = [ ];
+  test.identical( got, expected );
 
-  console.log('Hello world');
+  test.case = 'valid range with float step';
+  var got = _.longFromRangeWithStep( [ 0, 10 ], 1.5 );
+  var expected = [ 0, 1.5, 3, 4.5, 6, 7.5, 9 ];
+  test.identical( got, expected );
+
+  test.case = 'decrementing range with step < 0';
+  var got = _.longFromRangeWithStep( [ 10, 0 ], -1 );
+  var expected = [ 10, 9, 8, 7, 6, 5, 4, 3, 2, 1 ];
+  test.identical( got, expected );
+
+  test.case = 'step is undefined';
+  var got = _.longFromRangeWithStep( [ 4, 6 ], undefined );
+  var expected = [ 4, 5 ];
+  test.identical( got, expected );
+
+  test.case = 'step is undefined and first number bigger';
+  var got = _.longFromRangeWithStep( [ 10, 0 ], undefined );
+  var expected = [ 10, 9, 8, 7, 6, 5, 4, 3, 2, 1 ];
+  test.identical( got, expected );
+
+  test.case = 'decrementing range with step < 0';
+  var got = _.longFromRangeWithStep( [ 3, 2 ], -1 );
+  var expected = [ 3 ];
+  test.identical( got, expected );
+
+  /**/
+
+  if (!Config.debug)
+  return;
+
+  test.case = 'wrong arguments';
```

Comment:
Description of throwing test cases should be a more clear.

This test checks test routine called with single argument. And type of passed argument has not matter in this context. But for good test it should be a range

This test case looks like
test.case = 'not enough arguments';
test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 1, 10 ] ) );
The first argument is range, but routine calls without second argument.
//
Next test case checks validity of range
test.case = 'wrong type of range';
test.shouldThrowErrorSync( () => _.longFromRangeWithStep( 'wrong', 1 ) );
because, arguments.length === 2, and second argument is valid step. \
//
The same scheme for step argument 
test.case = 'wrong type of step';
test.shouldThrowErrorSync( () => _.longFromRangeWithStep( [ 5, 10 ], 'wrong' ) );
range is valid range, but step is not valid.